ariadne_codegen/client_generators/dependencies/base_client.py

```diff
@@ -70,9 +71,11 @@ def get_data(self, response: httpx.Response) -> dict[str, Any]:
     def _convert_dict_to_json_serializable(
         self, dict_: Dict[str, Any]
     ) -> Dict[str, Any]:
-        return {
-            key: value
-            if not isinstance(value, BaseModel)
-            else value.dict(by_alias=True)
-            for key, value in dict_.items()
-        }
+        def maybe_convert_element(path, key, value) -> Union[bool, Tuple[Any, Any]]:
+            if value is None:
+                return False
+            if isinstance(value, BaseModel):
+                return key, value.dict(by_alias=True)
+            else:
+                return key, value
+        return remap(dict_, maybe_convert_element)
```

Comment:
I wonder if it is worth introducing a new dependency, isn't it equivalent to something like this?
```py
def _convert_value(self, value):
    if isinstance(value, BaseModel):
        return value.dict(by_alias=True)

    if isinstance(value, list):
        return [self._convert_value(item) for item in value]

    return value


def _convert_dict_to_json_serializable(self, dict_: Dict[str, Any]) -> Dict[str, Any]:
    return {key: self._convert_value(value) for key, value in dict_.items()}
```

Reply:
The remap function is a bit of a hangover from another project where there were fewer guarantees on the types that were provided.

If we are guaranteed to only ever have combinations of `list`s and `pydantic.BaseModel`s then yes I think that the behaviour above is equivalent. If we might end up with `dict`s then the extra dependency is adding value.